tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -385,11 +385,15 @@ build_rootfs_distro()
 		mkdir -p ${ROOTFS_DIR}
 	fi
 
-	# need to detect rustc's version too?
-	detect_rust_version ||
-		die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-main}'."
-
-	echo "Required rust version: $RUST_VERSION"
+	# this check seems duplicate to the check in setup_rootfs, and we should
+	# also only need this check when we need to build the agent (see setup_rootfs)
+	if [ -z "${AGENT_SOURCE_BIN}" ] ; then
+		# need to detect rustc's version too?
+		detect_rust_version ||
+			die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-main}'."
+
+		echo "Required rust version: $RUST_VERSION"
+	fi
```

Comment:
I'd like to see this in a different commit, but still as part of the same PR.

But ... :-) ... can we try to simply remove this whole block? I think it's not needed.  And if it's needed, we should also check for AGENT_TARBALL.  But I truly believe it can be removed.

Reply:
To the best of my knowledge, this should not be needed. Happy to remove if nobody knows otherwise and if testing passes.

Reply:
Yay, let's remove it.

Reply:
I think we should remove this check for now indeed as there is no reason to check twice.

But: an early check in the case for when we need to build the agent may actually be reasonable. If you look at the relevant part in setup_rootfs(), the check looks a bit different, plus for instance a check for libseccomp is being executed (also, note that the .cargo related line is executed twice, unsure if that is required). Looks like that code can generally be improved.

Maybe in a subsequent PR one can think of adding a function that checks (and maybe installs) all the requirements early in order to fail early resp. more gracefully? Not sure what the intentions here are and whether it's worth the time. Thoughts? :)